music/src/core/player.ts

```diff
@@ -22,89 +22,116 @@ import {INoteSequence, NoteSequence} from '../protobuf/index';
-          resonance: 4000,
-          octaves: 1
-        })
-        .toMaster();
-const ride = new Tone.MetalSynth().toMaster();
-const crash = new Tone
-                  .MetalSynth({
-                    frequency: 300,
-                    envelope: {attack: 0.001, decay: 1, release: 3},
-                    harmonicity: 5.1,
-                    modulationIndex: 64,
-                    resonance: 4000,
-                    octaves: 1.5
-                  })
-                  .toMaster();
-const snare =
-    new Tone
-        .NoiseSynth({
-          noise: {type: 'white'},
-          envelope: {attack: 0.005, decay: 0.05, sustain: 0.1, release: 0.4}
-        })
-        .toMaster();
+  private closedHihat =
+      new Tone
+          .MetalSynth({
+            frequency: 400,
+            envelope: {attack: 0.001, decay: 0.1, release: 0.8},
+            harmonicity: 5.1,
+            modulationIndex: 32,
+            resonance: 4000,
+            octaves: 1
+          })
+          .toMaster();
+  private openHihat =
+      new Tone
+          .MetalSynth({
+            frequency: 400,
+            envelope: {attack: 0.001, decay: 0.5, release: 0.8, sustain: 1},
+            harmonicity: 5.1,
+            modulationIndex: 32,
+            resonance: 4000,
+            octaves: 1
+          })
+          .toMaster();
+  private ride = new Tone.MetalSynth().toMaster();
+  private crash = new Tone
+                      .MetalSynth({
+                        frequency: 300,
+                        envelope: {attack: 0.001, decay: 1, release: 3},
+                        harmonicity: 5.1,
+                        modulationIndex: 64,
+                        resonance: 4000,
+                        octaves: 1.5
+                      })
+                      .toMaster();
+  private snare =
+      new Tone
+          .NoiseSynth({
+            noise: {type: 'white'},
+            envelope: {attack: 0.005, decay: 0.05, sustain: 0.1, release: 0.4}
+          })
+          .toMaster();
 
-const drumKit: Array<(time: number) => void> = [
-  time => kick.triggerAttackRelease('C2', '8n', time),
-  time => snare.triggerAttackRelease('16n', time),
-  time => closedHihat.triggerAttack(time, 0.3),
-  time => openHihat.triggerAttack(time, 0.3),
-  time => tomLow.triggerAttack('G3', time, 0.5),
-  time => tomMid.triggerAttack('C4', time, 0.5),
-  time => tomHigh.triggerAttack('F4', time, 0.5),
-  time => crash.triggerAttack(time, 1.0), time => ride.triggerAttack(time, 0.5)
-];
+  private constructor() {
+    for (let c = 0; c < DEFAULT_DRUM_PITCH_CLASSES.length; ++c) {  // class
+      DEFAULT_DRUM_PITCH_CLASSES[c].forEach((p) => {
+        this.DRUM_PITCH_TO_CLASS.set(p, c);
+      });
+    }
+  }
+
+  static getInstance() {
+    if (!DrumKit.instance) {
+      DrumKit.instance = new DrumKit();
+    }
+    return DrumKit.instance;
+  }
 
+  public playNote(pitch: number, time: number) {
+    [() => this.kick.triggerAttackRelease('C2', '8n', time),
```

Comment:
Could this array be defined somewhere else (maybe as a member variable in the constructor)? I think as written, it will have to redefine the array for every call.

Reply:
Done